### PR TITLE
Added script and compiler option to run admin in debug mode

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
Signed-off-by: Prafulla Mahindrakar <prafulla.mahindrakar@gmail.com>

# TL;DR
Running admin in debug mode requires the compiler output to have debug bits.
Added a new Makefile target compile_debug which does this and the script debug.sh uses dlv to run the admin in debug mode for the serve option

Similarly the script can be modified to run for other commandline options exposed by flyteadmin

Pre-reequisite:
Install dlv  : https://github.com/go-delve/delve/blob/master/Documentation/installation/README.md
Setup IDE Debug configuration. The port for DLV should match
eg  
Screenshot 2021-04-15 at 3 30 10 PM

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

**Comment by kumare3** [comment]
Interesting can you debug the process in k8s? How? Do you have to start with delve

**Comment by pmahindrakar-oss** [comment]
@kumare3 this is only for admin running outside the k8 cluster. The locally running admin  in the IDE can use the k8 deployed services of minio and postgres using port forwarding . 

There is way to do the same for k8 deployed admin.
@EngHabu had suggested to use https://github.com/telepresenceio/telepresence/tree/release/v2 
I haven't tried that yet because the current way of running admin in the IDE works for me. 
We can try the telepresence route aswell if people are interested.
